### ISSUE TITLE
"Error: unable to read module BG/Q data from log file" in darshan-diff utility

### ISSUE THREAD

**Issue by shanedsnyder** [darshan-hpc/darshan#206]
In GitLab by @carns on Sep 18, 2016, 13:58

[pcarns_mpi-io-test_id12594_9-18-53686-17287241849796474061_1.darshan](/uploads/a44ac71b497cf3aad3d5c4e0cc530b5a/pcarns_mpi-io-test_id12594_9-18-53686-17287241849796474061_1.darshan)[pcarns_mpi-io-test_id12593_9-18-53685-12122538308779795328_1.darshan](/uploads/7111ed361bfa338339c253ec1f5c1677/pcarns_mpi-io-test_id12593_9-18-53685-12122538308779795328_1.darshan)

Files to reproduce are attached; these were generated with origin/master and two single-process runs of mpi-io-test.  Example:

```
pcarns@carns-x1:~$ darshan-diff pcarns_mpi-io-test_id12593_9-18-53685-12122538308779795328_1.darshan pcarns_mpi-io-test_id12594_9-18-53686-17287241849796474061_1.darshan
- # start_time: 1474224885
+ # start_time: 1474224886
- # end_time: 1474224885
+ # end_time: 1474224886
Error: unable to read module BG/Q data from log file.
Error: unable to build record hash for darshan log file pcarns_mpi-io-test_id12593_9-18-53685-12122538308779795328_1.darshan.
```

Relevant to issue #194 , need to make sure that darshan-diff works with stdio records.  I think it may be close already but there are a couple of weird things going on that don't seem to be stdio-related.

**Comment by shanedsnyder** [comment]
In GitLab by @carns on Sep 18, 2016, 14:15

Oh, it looks like this may be the same problem as described in #205.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Sep 19, 2016, 18:51

This issue should now be fixed in commits f21728f3 and ab32cf66. 

I tried it on the test files you have there and it works fine now.
